tests/apps/resiliencyapp/app.go

```diff
@@ -125,10 +141,11 @@ func resiliencyBindingHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	body, _ := io.ReadAll(r.Body)
 	var message FailureMessage
-	json.NewDecoder(r.Body).Decode(&message)
+	json.Unmarshal(body, &message)
```

Comment:
qq is this faster than decoder? why is this change needed?

Reply:
No, actually `json.NewDecoder(r.Body).Decode(&message)` is usually faster because it works on the stream.

The reason is that in the line below, I am now printing out the raw message (before it gets decoded, in case it's invalid JSON). Doing that requires loading the message in memory first (see `body, _ := io.ReadAll(r.Body)`). If the message isn't loaded in memory first, `NewDecoder(r.Body)` will consume the stream and cannot be read again.